sharktank/sharktank/ops/signatures.py

```diff
@@ -145,6 +150,38 @@ def _embedding_lookup_trampoline(
         d.fail(tensors)
 
 
+@overridable
+def equal(a: AnyTensor, b: AnyTensor) -> bool:
+    """Compares 2 tensors for equality, such that if one is substituted with the other
+    in sharktank polymorphic calls, the results will be essentially the same.
+    Meaning, they would also compare equal.
+
+    torch.Tensor and DefaultPrimitiveTensor with the same contents would compare equal."""
```

Comment:
Based on your trampoline, I think this needs a note like "overrides are matched first against both tensor types and failing that, then on just the first. Therefore, each override must internally decide whether it can handle an equality check with an arbitrary `b` tensor."